### PR TITLE
Fix uses of "int" where "int64" should be used instead

### PR DESCRIPTION
**Pull request by aaronlehmann**
Some structures use int for sizes and UNIX timestamps. On some
platforms, int is 32 bits, so this can lead to the year 2038 issues and
overflows when dealing with large containers or layers.

Consistently use int64 to store sizes and UNIX timestamps in
api/types/types.go. Update related to code accordingly (i.e.
strconv.FormatInt instead of strconv.Itoa).

Use int64 in progressreader package to avoid integer overflow when
dealing with large quantities. Update related code accordingly.

Signed-off-by: Aaron Lehmann aaron.lehmann@docker.com

### CODE REVIEW

**Comment by aaronlehmann** [comment]
Fixed `TestGetRemoteImageJSON`.

**Comment by aaronlehmann** [comment]
Rebased

**Comment by aaronlehmann** [comment]
Rebased again to rerun CI after /dev/urandom issues.

**Comment by aaronlehmann** [comment]
Rebased again. Can someone please review?

**Comment by LK4D4** [comment]
Makes sense, I hope nothing bad will happen with serialization
LGTM

**Comment by RichardScothern** [comment]
LGTM.  It might be nice to typedef a size type as int64, but I'm ok withs PR as is.

**Comment by icecrime** [comment]
LGTM

File: graph/graph_windows.go [hunk 1]
```diff
@@ -95,7 +95,7 @@ func (graph *Graph) storeImage(img *image.Image, layerData archive.ArchiveReader
 			}
 		}
 
-		if err := graph.saveSize(root, int(img.Size)); err != nil {
+		if err := graph.saveSize(root, img.Size); err != nil {
```
Comment (jlhawn) [inline graph/graph_windows.go]: I think this change is breaking the Windows CI?

Comment (jlhawn) [inline graph/graph_windows.go]: wait, no, my bad It's lower down.

Comment (jlhawn) [inline graph/graph_windows.go]: already taken care of too: https://github.com/docker/docker/pull/15318/files

**Pull request merged**